### PR TITLE
Update huawei & huaweicloud

### PR DESCRIPTION
**Pull request by KukiSa**
华为技术有限公司 粤A2-20044005号
华为云计算技术有限公司 黔ICP备20004760号

`abhouses.com` 粤A2-20044005号-28
`bisheng.cn` 粤A2-20044005号-100
`bishengcompiler.cn` 粤A2-20044005号-98
`gneec.com` 粤A2-20044005号-53
`gneec.com.cn` 粤A2-20044005号-55
`gneec3.com` 粤A2-20044005号-56
`gneec4.com` 粤A2-20044005号-57
`gneec7.com` 粤A2-20044005号-97
`hiascend.cn` 粤A2-20044005号-105
`hiascend.com` 粤A2-20044005号-107
`hiclc.com` 粤A2-20044005号-43
`hikunpeng.cn` 粤A2-20044005号-110
`hikunpeng.com` 粤A2-20044005号-108
`hikunpeng.com.cn` 粤A2-20044005号-111
`hikunpeng.net` 粤A2-20044005号-112
`hisilicon.com` 粤A2-20044005号-7
`hisilicon.com.cn` 粤A2-20044005号-24
`huawei.cn` 粤A2-20044005号-101
`huawei.com.cn` 粤A2-20044005号-30
`huaweiacad.com` 粤A2-20044005号-26
`huaweiief.cn` 粤A2-20044005号-73
`huaweiief.com` 粤A2-20044005号-60
`huaweistatic.com` 粤A2-20044005号-31
`hwccpc.cn` 粤A2-20044005号-61
`hwccpc.com` 粤A2-20044005号-38
`hwht.cn` 粤A2-20044005号-19
`hwocloud.com` 粤A2-20044005号-42
`hwtrip.com` 粤A2-20044005号-22
`mindspore.cn` 粤A2-20044005号-52
`owsgo.com` 粤A2-20044005号-63
`shanhaitujian.com` 粤A2-20044005号-113
`teleows.com` 粤A2-20044005号-62

`devui.design` 粤A2-20044005号-49
`huaweiapaas.com` 黔ICP备20004760号-35
`huaweicloud-smn.cn` 黔ICP备20004760号-95
`huaweicloud-smn.com` 黔ICP备20004760号-96
`huaweicloud-smn.net` 黔ICP备20004760号-97
`huaweicloudlive.com` 粤A2-20044005号-93
`huaweimarketplace.com` 黔ICP备20004760号-103
`huaweisafedns.cn` 粤A2-20044005号-84
`huaweisafehub.cn` 粤A2-20044005号-85
`huaweiyun.com` 粤A2-20044005号-59
`hwcdnsre.com` 黔ICP备20004760号-115
`hwclouds.cn` 黔ICP备20004760号-13
`hwclouds.mobi` 黔ICP备20004760号-13
`hwclouds.org.cn` 黔ICP备20004760号-13
`hwcloudvis.cn` 粤A2-20044005号-91
`hwcloudvis.com` 粤A2-20044005号-94
`imc-oneaccess.cn` 粤A2-20044005号-96
`imcapptest.com` 粤A2-20044005号-102
`km6g.cn` 黔ICP备20004760号-123
`km7g.cn` 黔ICP备20004760号-124
`km8g.cn` 黔ICP备20004760号-125
`km9g.cn` 黔ICP备20004760号-126
`livehwc3.cn` 粤A2-20044005号-76
`myhwcdn.cn` 粤A2-20044005号-77
`myhwcdn.com` 粤A2-20044005号-78
`myhwcloudlive.com` 粤A2-20044005号-92
`pagescube.cn` 黔ICP备20004760号-132
`pagescube.com` 黔ICP备20004760号-131
`saasops.tech` 粤A2-20044005号-47

`cdnhwchcg02.com` 黔ICP备20004760号-15
`cdnhwcurq03.com` 黔ICP备20004760号-16
`cdnhwczba04.com` 黔ICP备20004760号-17
`cdnhwclxw05.com` 黔ICP备20004760号-18
`cdnhwcupf06.com` 黔ICP备20004760号-19
`cdnhwcxcy07.com` 黔ICP备20004760号-20
`cdnhwcatq08.com` 黔ICP备20004760号-21
`cdnhwcztu09.com` 黔ICP备20004760号-22
`cdnhwcedi10.com` 黔ICP备20004760号-23
`cdnhwcoem01.com` 黔ICP备20004760号-24
`cdnhwcllh11.com` 黔ICP备20004760号-25
`cdnhwcjog12.com` 黔ICP备20004760号-26
`cdnhwcpsd13.com` 黔ICP备20004760号-27
`cdnhwcqwg14.com` 黔ICP备20004760号-28
`cdnhwcqir15.com` 黔ICP备20004760号-29
`cdnhwcvix16.com` 黔ICP备20004760号-30
`cdnhwcajk17.com` 黔ICP备20004760号-31
`cdnhwcchh18.com` 黔ICP备20004760号-32
`cdnhwcohm19.com` 黔ICP备20004760号-33
`cdnhwczjt20.com` 黔ICP备20004760号-34
`cdnhwcggk22.com` 黔ICP备20004760号-36
`cdnhwczth23.com` 黔ICP备20004760号-37
`cdnhwctxz24.com` 黔ICP备20004760号-38
`cdnhwcgrb01.cn` 黔ICP备20004760号-39
`cdnhwcyxb02.cn` 黔ICP备20004760号-40
`cdnhwcywp03.cn` 黔ICP备20004760号-41
`cdnhwclkl04.cn` 黔ICP备20004760号-42
`cdnhwcjbj05.cn` 黔ICP备20004760号-43
`cdnhwconz06.cn` 黔ICP备20004760号-44
`cdnhwcgqa21.com` 黔ICP备20004760号-45
`cdnhwckbw08.cn` 黔ICP备20004760号-46
`cdnhwcsjs09.cn` 黔ICP备20004760号-47
`cdnhwcjkl10.cn` 黔ICP备20004760号-48
`cdnhwctmz11.cn` 黔ICP备20004760号-49
`cdnhwcrpp12.cn` 黔ICP备20004760号-50
`cdnhwcpqh13.cn` 黔ICP备20004760号-51
`cdnhwcxsi14.cn` 黔ICP备20004760号-52
`cdnhwcfzp15.cn` 黔ICP备20004760号-53
`cdnhwcgvs16.cn` 黔ICP备20004760号-54
`cdnhwcrgg07.cn` 黔ICP备20004760号-55
`cdnhwceft18.cn` 黔ICP备20004760号-56
`cdnhwccvo19.cn` 黔ICP备20004760号-57
`cdnhwcbvo20.cn` 黔ICP备20004760号-58
`cdnhwcdkd21.cn` 黔ICP备20004760号-59
`cdnhwcdvg22.cn` 黔ICP备20004760号-60
`cdnhwclnh23.cn` 黔ICP备20004760号-61
`cdnhwckfd24.cn` 黔ICP备20004760号-62
`cdnhwczxh101.com` 黔ICP备20004760号-63
`cdnhwcbzj102.com` 黔ICP备20004760号-64
`cdnhwcjqv17.cn` 黔ICP备20004760号-65
`cdnhwcljk104.com` 黔ICP备20004760号-66
`cdnhwclxu105.com` 黔ICP备20004760号-67
`cdnhwcbqs106.com` 黔ICP备20004760号-68
`cdnhwctnm107.com` 黔ICP备20004760号-69
`cdnhwcbni108.com` 黔ICP备20004760号-70
`cdnhwczks109.com` 黔ICP备20004760号-71
`cdnhwcick110.com` 黔ICP备20004760号-72
`cdnhwcead111.com` 黔ICP备20004760号-73
`cdnhwcjlg112.com` 黔ICP备20004760号-74
`cdnhwckon103.com` 黔ICP备20004760号-75
`cdnhwczmn114.com` 黔ICP备20004760号-76
`cdnhwcqgw115.com` 黔ICP备20004760号-77
`cdnhwckfz116.com` 黔ICP备20004760号-78
`cdnhwcqve117.com` 黔ICP备20004760号-79
`cdnhwcgnc118.com` 黔ICP备20004760号-80
`cdnhwcuim119.com` 黔ICP备20004760号-81
`cdnhwcjsb120.com` 黔ICP备20004760号-82
`cdnhwccmz121.com` 黔ICP备20004760号-83
`cdnhwcibv122.com` 黔ICP备20004760号-84
`cdnhwcprh113.com` 黔ICP备20004760号-85
`cdnhwcedt124.com` 黔ICP备20004760号-86
`cdnhwcofl101.cn` 黔ICP备20004760号-87
`cdnhwclrd102.cn` 黔ICP备20004760号-88
`cdnhwcurr103.cn` 黔ICP备20004760号-89
`cdnhwczid104.cn` 黔ICP备20004760号-90
`cdnhwcqvk105.cn` 黔ICP备20004760号-91
`cdnhwczoy106.cn` 黔ICP备20004760号-92
`cdnhwcuwd107.cn` 黔ICP备20004760号-93
`cdnhwcogx108.cn` 黔ICP备20004760号-94
`cdnhwcoph123.com` 黔ICP备20004760号-98
`cdnhwceod109.cn` 黔ICP备20004760号-99
`cdnhwcpcw110.cn` 黔ICP备20004760号-100
`cdnhwczic111.cn` 黔ICP备20004760号-101
`cdnhwcozb112.cn` 黔ICP备20004760号-102
`cdnhwcqrr114.cn` 黔ICP备20004760号-104
`cdnhwcaoc115.cn` 黔ICP备20004760号-105
`cdnhwctyd116.cn` 黔ICP备20004760号-106
`cdnhwcidu117.cn` 黔ICP备20004760号-107
`cdnhwclsl119.cn` 黔ICP备20004760号-108
`cdnhwcsux118.cn` 黔ICP备20004760号-109
`cdnhwcbie120.cn` 黔ICP备20004760号-110
`cdnhwcuxc121.cn` 黔ICP备20004760号-111
`cdnhwcaip122.cn` 黔ICP备20004760号-112
`cdnhwcwqz113.cn` 黔ICP备20004760号-113
`cdnhwcojn124.cn` 黔ICP备20004760号-114
`cdnhwcscc123.cn` 黔ICP备20004760号-119

### CODE REVIEW

File: data/huaweicloud [hunk 1]
```diff
@@ -1,19 +1,47 @@
-cdnhwcibv122.com
+devui.design
```
Comment (rootmelo92118) [inline data/huaweicloud:1]: 希望可以新增一個huawei-dev引入huawei和category-dev

**Review by rootmelo92118 — CHANGES_REQUESTED** [review]

File: data/huaweicloud (hunk 1 above)
Comment (KukiSa) [reply data/huaweicloud]: 分离完成。

**Review by rootmelo92118 — APPROVED** [review]

File: data/huaweicloud [hunk 2]
```diff
@@ -40,5 +65,102 @@ na-mexico-1.myhuaweicloud.com @!cn      #拉美-墨西哥城一
 ru-northwest-2.myhuaweicloud.com @!cn   #俄罗斯-莫斯科二
 sa-brazil-1.myhuaweicloud.com @!cn      #拉美-圣保罗一
 
-regexp:.+\.cdnhwc([1-9]|10)\.cn$
-regexp:.+\.cdnhwc([1-9]|10)\.com$
+# 华为云 CDN CNAME 域名，按 ICP 备案序号排列
```
Comment (IceCodeNew) [inline data/huaweicloud:68]: 希望能提供查询备案信息的 URL，这样未来更新时也有个参考

Comment (KukiSa) [reply data/huaweicloud:68]: > 希望能提供查询备案信息的 URL，这样未来更新时也有个参考

全都是在 beian.miit.gov.cn 手工查的。

Comment (IceCodeNew) [reply data/huaweicloud:68]: > > 希望能提供查询备案信息的 URL，这样未来更新时也有个参考
> 
> 全都是在 beian.miit.gov.cn 手工查的。

辛苦了……
然后还要手动按照 ICP 备案序号排列吗，太可怕（感觉以后插入新域名很容易顺序就乱了 😝

Comment (KukiSa) [reply data/huaweicloud:68]: > > > 希望能提供查询备案信息的 URL，这样未来更新时也有个参考
> > 
> > 
> > 全都是在 beian.miit.gov.cn 手工查的。
> 
> 辛苦了…… 然后还要手动按照 ICP 备案序号排列吗，太可怕（感觉以后插入新域名很容易顺序就乱了 😝

`cdnhwc[a-z][a-z][a-z][0-9][0-9][0-9]?\.(cn|com)` 那一块（目前为第 69 至 164 行）为了方便维护还是按照备案序号排列吧，符合这一特征的域名有新增可以直接在后方加入。
维护本文件仅需不定期查询`黔ICP备20004760号-134`及以上的备案记录是否存在即可。
除开 `cdnhwc[a-z][a-z][a-z][0-9][0-9][0-9]?\.(cn|com)` 这一块，其他的域名按字母顺序排列我一般用 Excel 进行升序。

Comment (rootmelo92118) [reply data/huaweicloud:68]: > > > > 希望能提供查询备案信息的 URL，这样未来更新时也有个参考
> > > 
> > > 
> > > 全都是在 beian.miit.gov.cn 手工查的。
> > 
> > 
> > 辛苦了…… 然后还要手动按照 ICP 备案序号排列吗，太可怕（感觉以后插入新域名很容易顺序就乱了 😝
> 
> `cdnhwc[a-z][a-z][a-z][0-9][0-9][0-9]?\.(cn|com)` 那一块（目前为第 69 至 164 行）为了方便维护还是按照备案序号排列吧，符合这一特征的域名有新增可以直接在后方加入。 维护本文件仅需不定期查询`黔ICP备20004760号-134`及以上的备案记录是否存在即可。 除开 `cdnhwc[a-z][a-z][a-z][0-9][0-9][0-9]?\.(cn|com)` 这一块，其他的域名按字母顺序排列我一般用 Excel 进行升序。

`cdnhwc[a-z]{3}[0-9]{2,3}\.(cn|com)` 這樣感覺比較簡潔有力。

**Pull request merged**